module/observations/helper.js

```diff
@@ -61,77 +63,117 @@ module.exports = class ObservationsHelper {
+     * @param {String} requestingUserAuthToken - Requesting user auth token.
+     * @param {String} [programId = ""] - program id
      * @returns {Object} observation creation data.
      */
 
-    static create(solutionId, data, userDetails, requestingUserAuthToken = "") {
+    static create(
+        solutionId,
+        data, 
+        userId, 
+        requestingUserAuthToken = "",
+        programId = ""
+    ) {
         return new Promise(async (resolve, reject) => {
             try {
 
-                if(requestingUserAuthToken == "") {
+                if( requestingUserAuthToken == "" ) {
                     throw new Error(messageConstants.apiResponses.REQUIRED_USER_AUTH_TOKEN);
                 }
 
-                let userOrganisations = await shikshalokamHelper.getUserOrganisation(requestingUserAuthToken, userDetails.id);
-                let createdFor = new Array;
-                let rootOrganisations = new Array;
+                let organisationAndRootOrganisation = 
+                await shikshalokamHelper.getOrganisationsAndRootOrganisations(
+                    requestingUserAuthToken,
+                    userId
+                );
 
-                if(userOrganisations.success && userOrganisations.data) {
-                    createdFor = userOrganisations.data.organisations;
-                    rootOrganisations = userOrganisations.data.rootOrganisations;
-                } else {
-                    throw new Error(messageConstants.apiResponses.USER_ORGANISATION_DETAILS_NOT_FOUND);
-                }
+                let templateInformation = 
+                await solutionHelper.createProgramAndSolutionFromTemplate
+                (
+                    {
+                        _id : solutionId,
+                        type : messageConstants.common.OBSERVATION
+                    },
+                    userId,
+                    {
+                        id : programId
+                    }
+                );
 
-                let solutionDocument = await database.models.solutions.findOne({
-                    _id: ObjectId(solutionId),
-                    isReusable: true
-                }, {
-                        _id: 1,
-                        frameworkId: 1,
-                        frameworkExternalId: 1,
-                        externalId: 1,
-                        entityTypeId: 1,
-                        entityType: 1
-                    }).lean();
+                let observationData = 
+                await this.createObservation(
+                    data,
+                    userId,
+                    templateInformation.solution,
+                    organisationAndRootOrganisation
+                );
 
-                if (!solutionDocument) {
-                    throw new Error(messageConstants.apiResponses.SOLUTION_NOT_FOUND);
-                }
+                return resolve(_.pick(observationData, ["_id", "name", "description"]));
```

Comment:
@aman-tunerlabs - Please make sure response of old API to create observation doesn't change